web/views/api/helpers.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Castle.API.Helpers do
+
+  def meta_json(json, %{interval: interval, meta: meta}) do
+    json |> Map.merge(%{interval: interval, meta: meta})
+  end
+  def meta_json(json, %{meta: meta}) do
+    json |> Map.put(:meta, meta)
+  end
+
+  def counts_json(json, %{downloads: downloads}) do
+    json |> Map.put(:downloads, Enum.map(downloads, &count_json/1))
+  end
+  def counts_json(json, %{impressions: impressions}) do
+    json |> Map.put(:impressions, Enum.map(impressions, &count_json/1))
+  end
+
+  defp count_json(data), do: [data.time, data.count]
+
+  def groups_json(json, %{downloads: downloads}) do
+    json |> group_names(downloads) |> Map.put(:downloads, grouped(downloads))
+  end
+  def groups_json(json, %{impressions: impressions}) do
+    json |> group_names(impressions) |> Map.put(:impressions, grouped(impressions))
+  end
+
+  defp group_names(json, data) do
+    names = group_ranks(data) |> Enum.map(&group_find_rank(data, &1))
+    json |> Map.put(:groups, names)
+  end
+
+  defp group_ranks(data) do
+    data |> Enum.map(&(&1.rank)) |> Enum.uniq() |> Enum.sort()
+  end
+
+  defp group_find_rank(data, rank) do
+    Enum.find_value data, "Other", &(if &1.rank == rank do &1.display end)
+  end
+
+  defp grouped(raw_data) do
+    ranks = group_ranks(raw_data)
+    raw_data
```

Comment:
Hrmm... well, you could total them up yourself?  Or I could include the totals in these responses?

It's not actually any faster/cheaper for BQ to do a total-for-whole-interval rollup itself, vs giving you the smaller intervals back.  So the only thing it costs the client is a larger response size (but probably not large enough to matter even to a mobile client).

I was hoping the `/api/v1/podcasts/57` endpoint (and the episode one) would fulfill any "quick summary" needs we have.  Currently it just summarizes for some fixed intervals ... past-hour, past-day, etc.  But could make that smarter.